### PR TITLE
chore(wal): reduce memory allocation in tests

### PR DESCRIPTION
**Pull request by ideoma**
by decreasing O3 column memory buffers

### CODE REVIEW

**Comment by questdb-butler** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please update the PR title to match this format:
`type(subType): description`

Where `type` is one of:
`feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, `test`, `ci`, `chore`, `revert`

And: `subType` is one of:
`build`, `sql`, `log`, `mig`, `core`, `ilp`, `pgwire`, `http`, `conf`, `ui`, `wal`

For Example:

```
perf(sql): improve pattern matching performance for SELECT sub-queries
```
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fc3e74891a47901937ed33dbaa67336c979981cb
</p>

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/O3Utils.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|

File: core/src/main/java/io/questdb/cairo/O3Utils.java [hunk 1]
```diff
@@ -49,16 +49,20 @@ public static void setupWorkerPool(
...
         final int workerCount = workerPool.getWorkerCount();
         final O3PartitionPurgeJob purgeDiscoveryJob = new O3PartitionPurgeJob(messageBus, workerPool.getWorkerCount());
-        final ColumnPurgeJob columnPurgeJob = new ColumnPurgeJob(cairoEngine, functionFactoryCache);
 
-        workerPool.assign(purgeDiscoveryJob);
-        workerPool.assign(columnPurgeJob);
+        // ColumnPurgeJob has expensive init (it creates a table), disable it in some tests.
+        if (!cairoEngine.getConfiguration().disableColumnPurgeJob()) {
+            final ColumnPurgeJob columnPurgeJob = new ColumnPurgeJob(cairoEngine, functionFactoryCache);
+            workerPool.assign(purgeDiscoveryJob);
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/O3Utils.java:56]: should not be here

Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/O3Utils.java]: fixed in [45c5584](https://github.com/questdb/questdb/pull/2940/commits/45c558456860a7bc84357841539627f4321623e5)

**Review by bluestreak01 — APPROVED** [review]

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/O3Utils.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/O3Utils.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|

**Pull request merged**